glean-core/src/scheduler.rs

```diff
@@ -168,33 +248,67 @@ fn get_last_sent_build_metric() -> StringMetric {
 mod test {
     use super::*;
     use crate::tests::new_glean;
+    use std::cell::RefCell;
 
-    #[test]
-    fn test_different_app_builds() {
-        let (glean, _t) = new_glean(None);
-        let app_build = "a build".to_string();
-        let now = Local::now();
+    #[derive(Default)]
+    struct TestPolicy {
+        app_build: String,
+        last_sent_build: RefCell<Option<String>>,
+        last_sent_time: RefCell<Option<DateTime<FixedOffset>>>,
+        last_sleep_duration: RefCell<Option<Duration>>,
+        last_generation: RefCell<Option<u32>>,
+        last_reason: RefCell<Option<&'static str>>,
+    }
 
-        get_last_sent_build_metric().set(&glean, "a different build");
+    impl Policy for TestPolicy {
+        fn app_build(&self, _glean: &Glean) -> String {
+            self.app_build.to_string()
+        }
 
-        schedule_internal(&glean, &app_build, now);
+        fn get_last_sent_build(&self, _glean: &Glean) -> Option<String> {
+            self.last_sent_build.borrow().clone()
+        }
+        fn set_last_sent_build(&self, _glean: &Glean, value: &str) {
+            self.last_sent_build.replace(Some(value.to_string()));
+        }
+        fn get_last_sent_time(&self, _glean: &Glean) -> Option<DateTime<FixedOffset>> {
+            *self.last_sent_time.borrow()
+        }
+        fn set_last_sent_time(&self, _glean: &Glean, value: DateTime<Local>) {
+            self.last_sent_time.replace(Some(value.into()));
+        }
 
-        assert_eq!(
-            Some(app_build),
-            get_last_sent_build_metric().get_value(&glean, INTERNAL_STORAGE)
-        );
+        fn schedule_on_thread(
+            &self,
+            sleep_duration: Duration,
+            generation: u32,
+            reason: Option<&'static str>,
+        ) {
+            self.last_sleep_duration.replace(Some(sleep_duration));
+            self.last_generation.replace(Some(generation));
+            self.last_reason.replace(reason);
+        }
     }
 
     #[test]
-    fn test_case_1() {
+    fn test_different_app_builds() {
         let (glean, _t) = new_glean(None);
-        let now = Local.ymd(2021, 4, 28).and_hms(14, 57, 15);
-        let last_sent = Local.ymd(2021, 4, 28).and_hms(4, 0, 0);
+        let mut policy = TestPolicy {
+            ..Default::default()
+        };
+        policy.app_build = "a build".to_string();
+        policy
+            .last_sent_build
+            .replace(Some("a different build".to_string()));
```

Comment:
This can probably be simplified to:

```rust
        let mut policy = TestPolicy {
			app_buid: "a build".to_string(),
			last_sent_build: Some("a different build".to_string)),
            ..Default::default()
        };
```